src/material/paginator/paginator.ts

```diff
@@ -199,6 +201,56 @@ export class MatPaginator implements OnInit, OnDestroy {
   /** Emits when the paginator is initialized. */
   initialized: Observable<void> = this._initializedStream;
 
+  /** Element references for paginator buttons */
+  @ViewChild('nextButton', {read: ElementRef, static: false})
+  nextButtonRef: ElementRef;
+  @ViewChild('previousButton', {read: ElementRef, static: false})
+  previousButtonRef: ElementRef;
+  @ViewChild('firstButton', {read: ElementRef, static: false})
+  firstButtonRef: ElementRef;
+  @ViewChild('lastButton', {read: ElementRef, static: false})
+  lastButtonRef: ElementRef;
+
+  /** Changes focus to next appropriate button whenever current button becomes disabled. */
+  handleButtonBlur(buttonRef: MatIconButton) {
```

Comment:
Rather than moving focus around when the button is blurred, we can make the disabled buttons focusable instead. If you set the `disabledInteractive` input on the button, it'll be `aria-disabled="true"` instead of actually being disabled which won't blur it.

Reply:
added disabledInteractive to buttons, this prevents the focus from going to the body whenever a button becomes disabled. Thanks!